pilot/pkg/serviceregistry/serviceentry/servicediscovery.go

```diff
@@ -791,7 +791,15 @@ func (s *ServiceEntryStore) GetProxyServiceInstances(node *model.Proxy) []*model
 	for _, ip := range node.IPAddresses {
 		instances, found := s.ip2instance[ip]
 		if found {
-			out = append(out, instances...)
+			for _, i := range instances {
+				// Insert all instances for this IP for services within the same namespace This ensures we
```

Comment:
How does it relate to Gateway ?

Reply:
I mean - we have the general problem that gateway label selection is cross namespce.

Reply:
Already merged in another PR, will remove. But the reason is for the GatewayService stuff - we don't want them picking a service in another namespace